File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/processing/ArrayVisitor.java
package net.dzikoysk.openapi.processor.processing;

import javax.lang.model.element.AnnotationValue;
import java.util.List;
import java.util.stream.Collectors;

public final class ArrayVisitor<T> extends DefaultVisitor<List<? extends T>, Void> {

    @Override
    @SuppressWarnings("unchecked")
    public List<? extends T> visitArray(List<? extends AnnotationValue> values, Void unused) {
        return values.stream()
                .map(value -> (T) value.getValue())
                .collect(Collectors.toList());
    }

}

File: openapi-annotation-processor/src/main/kotlin/io/javalin/openapi/processor/processing/ArrayVisitor.kt
package io.javalin.openapi.processor.processing

import javax.lang.model.element.AnnotationValue

internal class ArrayVisitor<T> : DefaultVisitor<List<T>, Void?>() {

    @Suppress("UNCHECKED_CAST", "PARAMETER_NAME_CHANGED_ON_OVERRIDE")
    override fun visitArray(values: List<AnnotationValue>, unused: Void?): List<T> =
        values.map { value: AnnotationValue -> value.value as T }

}